src/libraries/Common/src/System/Diagnostics/DiagnosticsHelper.cs

```diff
@@ -7,62 +7,241 @@ namespace System.Diagnostics
 {
     internal static class DiagnosticsHelper
     {
-        internal static bool CompareTags(IEnumerable<KeyValuePair<string, object?>>? tags1, IEnumerable<KeyValuePair<string, object?>>? tags2)
+        // This is similar to System.Linq ToArray. We are not using System.Linq here to avoid the dependency.
+        internal static KeyValuePair<string, object?>[]? ToArray(IEnumerable<KeyValuePair<string, object?>>? tags)
```

Comment:
If we really can't use Enumerable.ToArray for some reason, can we just use `new List<T>(enumerable)` or `new List<T>(enumerable).ToArray()`?

Reply:
I think it is possible. I had it as a List<> before but I got a suggestion to make it array to now allow growing the list. I think it will be ok to revert using List<T>. 

CC @noahfalk

Reply:
List is fine. Array suggestion was just a minor perf optimization.